core/GlobalState.cc

```diff
@@ -1520,21 +1521,21 @@ NameRef GlobalState::freshNameUnique(UniqueNameKind uniqueNameKind, NameRef orig
         }
     }
 
+    auto name = NameRef(*this, NameKind::UNIQUE, names.size());
     auto &bucket = namesByHash[bucketId];
     bucket.first = hs;
-    bucket.second = names.size();
+    bucket.second = name.rawId();
 
-    auto idx = names.size();
-    names.emplace_back();
+    auto &nameData = names.emplace_back();
 
-    names[idx].kind = NameKind::UNIQUE;
-    names[idx].unique.num = num;
-    names[idx].unique.uniqueNameKind = uniqueNameKind;
-    names[idx].unique.original = original;
-    ENFORCE(NameRef(*this, idx).hash(*this) == hs);
+    nameData.kind = NameKind::UNIQUE;
+    nameData.unique.num = num;
+    nameData.unique.uniqueNameKind = uniqueNameKind;
+    nameData.unique.original = original;
+    ENFORCE(name.hash(*this) == hs);
```

Comment:
Would it be worth adding a constructor to `Name` for this case, so that you can just `emplace_back` the arguments to it?

Reply:
`Name` is going away real soon, so I'm not terribly invested in improving its ergonomics.